web/src/features/charts/futurePriceUtils.ts

```diff
@@ -72,3 +72,30 @@ export const negativeToPostivePercentage = (
 
   return Math.round(Math.abs((minPrice / (maxPrice + Math.abs(minPrice))) * 100));
 };
+
+export const calculateWidth = (
+  width: number | null | undefined,
+  negativePercentage: number,
+  isNegative: boolean
+): number => {
+  if (!width) {
+    return 0;
+  }
+  if (isNegative) {
+    return ((width - 12) * negativePercentage) / 100 + 12;
+  }
+  return ((width - 12) * (100 - negativePercentage)) / 100 + 12;
```

Comment:
Would be nice with some comments that describe what these magic numbers are for our future selfs, or extracting them to constants as they seem to be re-used a lot.